packages/components/src/custom-select-control-v2/test/index.tsx

```diff
@@ -13,493 +13,35 @@ import { useState } from '@wordpress/element';
-
-			it( 'Can change selection with a focused input and closed dropdown if typed characters match an option', async () => {
-				render( <Component { ...legacyProps } /> );
-
-				const currentSelectedItem = screen.getByRole( 'combobox', {
-					expanded: false,
-				} );
-
-				await sleep();
-				await press.Tab();
-				expect( currentSelectedItem ).toHaveFocus();
-
-				await type( 'aq' );
-
-				expect(
-					screen.queryByRole( 'listbox', {
-						name: 'label!',
-						hidden: true,
-					} )
-				).not.toBeInTheDocument();
-
-				await press.Enter();
-				expect( currentSelectedItem ).toHaveTextContent( 'aquamarine' );
-			} );
-
-			it( 'Should have correct aria-selected value for selections', async () => {
-				render( <Component { ...legacyProps } /> );
-
-				const currentSelectedItem = screen.getByRole( 'combobox', {
-					expanded: false,
-				} );
-
-				await click( currentSelectedItem );
-
-				// get all items except for first option
-				const unselectedItems = legacyProps.options.filter(
-					( { name } ) => name !== legacyProps.options[ 0 ].name
-				);
-
-				// assert that all other items have aria-selected="false"
-				unselectedItems.map( ( { name } ) =>
-					expect(
-						screen.getByRole( 'option', { name, selected: false } )
-					).toBeVisible()
-				);
-
-				// assert that first item has aria-selected="true"
-				expect(
-					screen.getByRole( 'option', {
-						name: legacyProps.options[ 0 ].name,
-						selected: true,
-					} )
-				).toBeVisible();
-
-				// change the current selection
-				await click( screen.getByRole( 'option', { name: 'poppy' } ) );
-
-				// click combobox to mount listbox with options again
-				await click( currentSelectedItem );
-
-				// check that first item is has aria-selected="false" after new selection
-				expect(
-					screen.getByRole( 'option', {
-						name: legacyProps.options[ 0 ].name,
-						selected: false,
-					} )
-				).toBeVisible();
-
-				// check that new selected item now has aria-selected="true"
-				expect(
-					screen.getByRole( 'option', {
-						name: 'poppy',
-						selected: true,
-					} )
-				).toBeVisible();
-			} );
-		} );
-	} );
-} );
-
-describe( 'static typing', () => {
-	<>
-		{ /* @ts-expect-error - when `options` prop is passed, `onChange` should have legacy signature */ }
-		<UncontrolledCustomSelect
-			label="foo"
-			options={ [] }
-			onChange={ ( _: string | string[] ) => undefined }
-		/>
-		<UncontrolledCustomSelect
-			label="foo"
-			options={ [] }
-			onChange={ ( _: { selectedItem: unknown } ) => undefined }
-		/>
-		<UncontrolledCustomSelect
-			label="foo"
-			onChange={ ( _: string | string[] ) => undefined }
-		>
-			foobar
-		</UncontrolledCustomSelect>
-		{ /* @ts-expect-error - when `children` are passed, `onChange` should have new default signature */ }
-		<UncontrolledCustomSelect
-			label="foo"
-			onChange={ ( _: { selectedItem: unknown } ) => undefined }
-		>
-			foobar
-		</UncontrolledCustomSelect>
-	</>;
-} );
```

Comment:
I removed these static types tests altogether, because they were only necessary for verifying the complex union type behavior in the legacy adapter layer.

Now that the types are straightforward, we don't need to test them.

Reply:
Similar to above, it makes sense to remove the `With Default Props` test group now that we test only the v2 props anyway.